### PR TITLE
Add --all option to plugins docs and also explain downgrading with an example

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title`, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result)
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

tpmccallum@tpmccallum:~/developer$ bart check content/spin/v2/managing-plugins.md 
✅ content/spin/v2/managing-plugins.md

- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/spin/v2/managing-plugins.md [hunk 1]
```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+## Refreshing the Catalogue
+
+The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 
+
+To update your local cache of the catalogue, run `spin plugins update`.
+
 ## Upgrading Plugins
 
-To upgrade a plugin to the latest version, run `spin plugins upgrade`. This has the same options as `spin plugins install`, according to whether the plugin comes from the catalogue, a URL, or a file:
+To upgrade a plugin to the latest version, first run `spin plugins upgrade` (to refresh the catalogue).
```
Comment (karthik2804) [inline content/spin/v2/managing-plugins.md:128]:
```suggestion
To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue). 
```

File: content/spin/v2/managing-plugins.md [hunk 2]
```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
 $ spin plugins upgrade js2wasm
+```
+
+The following example shows how to upgrade all installed plugins at once:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin plugins update
+$ spin plugins --all
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:149]:
```suggestion
$ spin plugins upgrade --all
```

File: content/spin/v2/managing-plugins.md [hunk 3]
```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+By default, Spin will only _upgrade_ plugins. Pass the `--downgrade` flag and specify the `--version` if you want Spin to roll back to an earlier version. The following abridged example (which doesn't list the full console output for simplicity) lists the versions of plugins, downgrades the `js2wasm` to an older version (`0.6.0`) and then lists the versions again to show the results:
 
-## Refreshing the Catalogue
+<!-- @nocpy -->
 
-The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 
+```bash
+$ spin plugins update
+$ spin plugins list
+js2wasm 0.1.0
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:170]: Consider shortening this list - it overwhelmes the line you actually want to show.

(You could also add blank lines above commands to make them stand out from the console output, but I'm not sure if that will be needed if you truncate the output.)

File: content/spin/v2/managing-plugins.md [hunk 4]
```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+js2wasm 0.3.0
+js2wasm 0.4.0
+js2wasm 0.5.0
+js2wasm 0.5.1
+js2wasm 0.6.0 [installed]
+js2wasm 0.6.1
+```
 
-To update your local cache of the catalogue, run `spin plugins update`.
+After the success of the downgrade process, we can see that the `[installed]` indicator is now aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:190]: Suggest simplifying the sentence structure

```suggestion
After downgrading, that the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```

File: content/spin/v2/managing-plugins.md [hunk 5]
```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+The following example shows how to upgrade one plugin at a time (i.e. the `js2wasm` plugin):
 
 <!-- @selectiveCpy -->
 
 ```bash
+$ spin plugins update
 $ spin plugins upgrade js2wasm
+```
+
+The following example shows how to upgrade all installed plugins at once:
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:143]: This only installs plugins from the registry - do we need to express that explicitly?

File: content/spin/v2/managing-plugins.md (hunk 3 above)
Comment (tpmccallum) [reply content/spin/v2/managing-plugins.md:170]: Thanks, will snip

File: content/spin/v2/managing-plugins.md (hunk 5 above)
Comment (tpmccallum) [reply content/spin/v2/managing-plugins.md:143]: Good point, will add that for clarity.

File: content/spin/v2/managing-plugins.md [hunk 6]
```diff
@@ -116,27 +117,69 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+## Refreshing the Catalogue
+
+The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 
+
+To update your local cache of the catalogue, run `spin plugins update`.
+
 ## Upgrading Plugins
 
-To upgrade a plugin to the latest version, run `spin plugins upgrade`. This has the same options as `spin plugins install`, according to whether the plugin comes from the catalogue, a URL, or a file:
+To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue).
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:128]:
```suggestion
To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue), then `spin plugins upgrade`. 
```

File: content/spin/v2/managing-plugins.md [hunk 7]
```diff
@@ -116,27 +117,69 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
...
+js2wasm 0.6.1 [installed]
+$ spin plugins upgrade js2wasm --downgrade --version 0.6.0
+$ spin plugins list
+// --snip--
+js2wasm 0.6.0 [installed]
+js2wasm 0.6.1
+```
 
-To update your local cache of the catalogue, run `spin plugins update`.
+After downgrading, that the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```
Comment (itowlson) [inline content/spin/v2/managing-plugins.md:182]:
```suggestion
After downgrading, the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```

**Review by itowlson — APPROVED** [review]

**Pull request merged**